frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1120,6 +1077,71 @@ impl<T: Config> ElectionProvider<T::AccountId, T::BlockNumber> for Pallet<T> {
 	}
 }
 
+fn maybe_trim<T>(items: Vec<T>, maybe_size: Option<usize>) -> Vec<T> {
+	if let Some(size) = maybe_size {
+		items.into_iter().take(size).collect::<Vec<T>>()
+	} else {
+		items
+	}
+}
```

Comment:
I find the `size` name a bit confusing. The implementation can also avoid allocating a new vector:

```suggestion
fn maybe_trim<T>(mut items: Vec<T>, maybe_qty: Option<usize>) -> Vec<T> {
    items.truncate(maybe_qty.unwrap_or_else(|| items.len()));
    items
}
```